src/vmm/src/builder.rs

```diff
@@ -566,11 +566,17 @@ pub fn build_microvm_from_snapshot(
     vm_resources.update_vm_config(&VmUpdateConfig {
         vcpu_count: Some(vcpu_count),
         mem_size_mib: Some(mem_size_mib(&guest_memory) as usize),
-        smt: Some(false),
-        cpu_template: None,
+        smt: Some(microvm_state.vm_info.smt),
+        cpu_template: Some(
```

Comment:
I think this non-ideal. It introduces a large dependency (serde) and complex process (serialization) where its not really needed.
```rust
serde_json::from_str(microvm_state.vm_info.cpu_template.as_str()).unwrap_or_default()
// vs
CpuFeaturesTemplate::try_from(microvm_state.vm_info.cpu_template.as_str()).unwrap_or_default()
```
Underneath serde is doing the same thing as match, just less clearly in my opinion.

Reply:
I would disagree here, I think this adds additional complexity by pulling in a dependency.

Reply:
CpuFeaturesTemplate already implements Deserialize, there is not extra dependency added.
The suggested code should compile out of the box.
Are you referring to smth else when mentioning dependency>

Reply:
I am referencing that this specific code would require the dependency of `serde_json` being present and `CpuFeaturesTemplate` implementing `deserialize`. I think by using `serde_json` here this specific area of code becomes more difficult to understand (it may take a moment to consider it is just matching the string, it using quite complex tools for something very simple). 

With `String::try_from` there is less indirection as someone can immediately jump to the code to see what is occurring. In both cases the code does the same thing, but with `String::try_from` I think it is simpler for someone to see.

Reply:
I think you make a very good point and I do not think it would have made any sense to use `serde_json` here if it were not for `CpuFeaturesTemplate` already having a dependency to it: https://github.com/firecracker-microvm/firecracker/blob/196ac5a4c38adb9dcd8068c02c93c2a22ba8b726/src/vmm/src/vmm_config/machine_config.rs#L240.

We need to be aware that `CpuFeaturesTemplate` is the structure that we use for transposing API user requests into Cpu Templates so it will always have a dependency on serde.
As such, I believe adding new code for implementing `TryFrom` for `CpuFeaturesTemplate` is not valuable for this specific context, It will add code that needs to be unit tested and maintained without actually having a real need for it.

Reply:
In looking this up it seems there is a specific trait for this use case [`std::str::FromStr`](https://doc.rust-lang.org/std/str/trait.FromStr.html) like https://stackoverflow.com/a/61948093/4301453. Does this seem a better approach?

Reply:
I could store `CpuFeaturesTemplate` itself as part of the VmInfo instead of a `String`
This would remove the need to convert to and from string altogether. WDYT?

Reply:
I like it, yes @vibharya

Reply:
I wasn't sure if this was possible, but this would be better yes.